test/unit-test/mqtt_agent_utest.c

```diff
@@ -74,21 +110,455 @@ int suiteTearDown( int numFailures )
     return numFailures;
 }
 
+/* ========================================================================== */
 
 /**
- * @brief Test that MQTTAgent_Init is able to update the context object correctly.
+ * @brief A mocked send function to send commands to the agent.
  */
+static bool mockSend( AgentMessageContext_t * pMsgCtx,
+                      const void * pData,
+                      uint32_t blockTimeMs )
+{
+    Command_t ** pCommandToSend = ( Command_t ** ) pData;
 
-void test_MQTTAgent_Init_Happy_Path( void )
+    pMsgCtx->pSentCommand = *pCommandToSend;
+    return true;
+}
+
+/**
+ * @brief A mocked send function to send commands to the agent.
+ * Returns failure.
+ */
+static bool mockSendFail( AgentMessageContext_t * pMsgCtx,
+                          const void * pData,
+                          uint32_t blockTimeMs )
 {
+    ( void ) pMsgCtx;
+    ( void ) pData;
+    ( void ) blockTimeMs;
+    return false;
+}
 
+/**
+ * @brief A mocked receive function for the agent to receive commands.
+ */
+static bool mockReceive( AgentMessageContext_t * pMsgCtx,
+                         void * pBuffer,
+                         uint32_t blockTimeMs )
+{
+    Command_t ** pCommandToReceive = ( Command_t ** ) pBuffer;
 
+    *pCommandToReceive = pMsgCtx->pSentCommand;
+    return true;
+}
+
+/**
+ * @brief A mocked function to obtain an allocated command.
+ */
+static Command_t * mockGetCommand( uint32_t blockTimeMs )
+{
+    return pCommandToReturn;
+}
+
+/**
+ * @brief A mocked function to release an allocated command.
+ */
+static bool mockReleaseCommand( Command_t * pCommandToRelease )
```

Comment:
same

Reply:
also for the rest of them
some arguably might be called fake since they do some processing ( if, ++ ), maybe not enough